entname/entname.go

```diff
@@ -211,6 +211,41 @@ func Fabric(dp *DeviceParams, index int) (string, error) {
 	return namer.Fabric(uint(index))
 }
 
+// CommonTrafficQueueNames are the names of common traffic class queues.
+type CommonTrafficQueueNames struct {
+	NC1, AF4, AF3, AF2, AF1, BE1, BE0 string
+}
+
+func (qn *CommonTrafficQueueNames) String() string {
+	if qn == nil {
+		return "nil"
+	}
+	return fmt.Sprintf("%+v", *qn)
+}
+
+// CommonTrafficQueues returns the vendors-specific names of common traffic
+// class queues. See the definition of common queues here:
+// https://github.com/openconfig/entity-naming/blob/main/README.md#traffic-queues
+func CommonTrafficQueues(dev *DeviceParams) (*CommonTrafficQueueNames, error) {
+	namer, err := lookupNamer(dev)
+	if err != nil {
+		return nil, err
+	}
+	tcq, err := namer.CommonTrafficQueues()
+	if err != nil {
+		return nil, err
+	}
+	return &CommonTrafficQueueNames{
+		NC1: tcq.NC1,
+		AF4: tcq.AF4,
+		AF3: tcq.AF3,
+		AF2: tcq.AF2,
+		AF1: tcq.AF1,
+		BE1: tcq.BE1,
+		BE0: tcq.BE0,
+	}, nil
```

Comment:
One reason is to avoid the circular dependency. The other is a little-bit future proofing that the namer.CommonTrafficQueueNames could become more restricted in ways that are helpful to implementors whereas the user-facing entname.CommonTrafficQueueNames should stay user-friendly for test authors. See how entname.PortParams and namer.PortParams differ for an example of how this distinction could in theory be useful.

In the Signal example, both the unix.Signal and syscall.Signal are both exported. Doing that results in two user-facing types and the risk one uses the one we didn't want them to use. It also means an extra hop to find the documentation. If the new package is internal, well, I'd have to test that to check that it would work, but it would still require the documentation hop and perhaps be confusing if the details live within an internal package.

On balance, I think I'd like to stick with the way it is, because (1) the duplication is not a huge cost; (2) there are some downsides to creating a new package just for this struct; and (3) it offers some flexibility for evolving the implementation-facing struct independently from the user-facing one.